routes/notes.js

```diff
@@ -1,50 +1,79 @@
 'use strict';
 
+const Note = require('../models/note');
 const express = require('express');
-
 const router = express.Router();
+const mongoose = require('mongoose');
 
 /* ========== GET/READ ALL ITEMS ========== */
 router.get('/', (req, res, next) => {
-
-  console.log('Get All Notes');
-  res.json([
-    { id: 1, title: 'Temp 1' },
-    { id: 2, title: 'Temp 2' },
-    { id: 3, title: 'Temp 3' }
-  ]);
-
+  const {searchTerm, folderId} = req.query;
+
+  let filter = {};
+  if (searchTerm) {
+    const re = new RegExp(searchTerm, 'i');
+    filter = {$or: [{ 'title': re }, { 'content': re }] };
+  }
+  if (folderId) {
+    filter = {folderId};
+  }
+  Note.find(filter)
+    .sort({ updatedAt: 'desc' })
+    .then(results => {
+      res.json(results);
+    })
+    .catch(err =>  next(err));
 });
 
 /* ========== GET/READ A SINGLE ITEM ========== */
 router.get('/:id', (req, res, next) => {
-
-  console.log('Get a Note');
-  res.json({ id: 1, title: 'Temp 1' });
-
+  const id = req.params.id;
+  Note.findById(id)
```

Comment:
Validate id before querying the database
Populate tags (required) and folders (optional)
handle id not found by calling next()